File: 18 - Binary Gap Length/solution.py
def binary_gap_length(n):
    binary = bin(n)
    gaps = binary.split("1") # Split the binary string into an array of gaps
    gaps.pop() # Remove the last gap

    max_gap = 0
    for gap in gaps:
        max_gap = max(max_gap, len(gap))

    return max_gap


# Test Cases
print(binary_gap_length(9))
print(binary_gap_length(10))        
print(binary_gap_length(529))

File: 18 - Binary Gap Length/solution.js
function binaryGapLength(n) {
  const binary = n.toString(2);
  const gaps = binary.split("1"); // Split the binary string into an array of gaps
  gaps.pop(); // Remove the last gap

  let maxGap = 0;
  for (let i = 0; i < gaps.length; i++) {
    maxGap = Math.max(maxGap, gaps[i].length);
  }

  return maxGap;
}

// Test Cases
console.log(binaryGapLength(9));
console.log(binaryGapLength(10));
console.log(binaryGapLength(529));
